src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -111,48 +123,100 @@ static CallingConvention getCallingConvention(FunctionPointerCallingConventionSy
+                                    customModifiers.Add(handleIndividualUnrecognizedSpecifier(specifier, typeBinder, diagnostics));
                                 }
 
-                                return CallingConvention.Default;
+                                return CallingConvention.Unmanaged;
                         }
 
                     case var unexpected:
                         throw ExceptionUtilities.UnexpectedValue(unexpected);
                 }
 
-                static CallingConvention reportBadConventionAndReturn(string givenConvention, SyntaxNodeOrToken node, DiagnosticBag diagnostics)
+                static CallingConvention handleSingleConvention(FunctionPointerUnmanagedCallingConventionSyntax specifier, Binder typeBinder, ArrayBuilder<CustomModifier> customModifiers, DiagnosticBag diagnostics)
+                {
+                    checkUnmanagedSupport(typeBinder, specifier.GetLocation(), diagnostics, isExtensionError: true);
+                    customModifiers.Add(handleIndividualUnrecognizedSpecifier(specifier, typeBinder, diagnostics));
+                    return CallingConvention.Unmanaged;
+                }
+
+                static CustomModifier handleIndividualUnrecognizedSpecifier(FunctionPointerUnmanagedCallingConventionSyntax specifier, Binder typeBinder, DiagnosticBag diagnostics)
+                {
+                    const string CallingConventionMetadataFormatString = "System.Runtime.CompilerServices.CallConv{0}";
+                    string specifierText = specifier.Name.ValueText;
+                    var metadataName = string.Format(CultureInfo.InvariantCulture, CallingConventionMetadataFormatString, specifierText);
+
+                    var specifierType = typeBinder.Compilation.Assembly.CorLibrary.GetTypeByMetadataName(metadataName, includeReferences: false, isWellKnownType: true, conflicts: out _, warnings: null);
+
+                    if (specifierType is null)
+                    {
+                        // Type name is CallConv{0}
+                        specifierType = typeBinder.CreateErrorType(name: metadataName[32..]);
+                        diagnostics.Add(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, specifier.GetLocation(), specifierText);
+                    }
+
+                    return CSharpCustomModifier.CreateOptional(specifierType);
+                }
+
+                static void checkUnmanagedSupport(Binder typeBinder, Location errorLocation, DiagnosticBag diagnostics, bool isExtensionError)
                 {
-                    if (!node.ContainsDiagnostics)
+                    if (!typeBinder.Compilation.Assembly.RuntimeSupportsUnmanagedSignatureCallingConvention)
                     {
-                        diagnostics.Add(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, node.GetLocation(), givenConvention);
+                        diagnostics.Add(isExtensionError ? ErrorCode.ERR_RuntimeDoesNotSupportUnmanagedExtensionCallConv
```

Comment:
>ERR_RuntimeDoesNotSupportUnmanagedExtensionCallConv [](start = 69, length = 51)

I am not sure I see a benefit from differentiating the error. Since we check the same runtime support flag, the both scenarios are either supported, or not supported at the same time.